README.md

```diff
@@ -48,59 +49,168 @@ For .NET Core CLI, use the following command:
 Install-Package TagShelf.Alfred.ApiWrapper
 ```
 
-## Usage
+# Environment
+
+You will always use the production environment for all your integrations using this library. The production environment is used for live applications and real-world scenarios. It is recommended to use this environment for production-ready applications. The frontend URL for the production environment is [https://app.tagshelf.com](https://app.tagshelf.com)</br></br>
+
+# Authentication
+
+Alfred supports three authentication methods: OAuth, HMAC, and App API Key. Each method is suited for different scenarios. To obtain the necessary credentials for the following authentication methods, please refer to the [Alfred API documentation](https://docs.tagshelf.dev/authentication) or contact the Alfred support team.
+
+|OAuth||
+|--|--|
+| **When to Use**   | Ideal for web portals needing integration with document parsers or API-driven services, especially when users access through browsers. Suitable for short-lived browser sessions and scenarios requiring scoped, secure access.|
+| **Benefits**      | Enhanced security via token use, flexible access control, and a user-friendly experience with periodic re-authentication.|
+| **Best Practices**| Secure token storage and graceful handling of token expiration.|
+
+To initialize the client for OAuth authentication, specify the credentials explicitly:
```

Comment:
specify the corresponding username and password.

(Eduardo's comment: since this section uses the word credential for many things, we need to be very specific about which credential we are talking about.)

Reply:
Edited so that the user knows its their tagshelf credentials. As I undestand it they only have one set of creds to access "Alfred", those are the creds needed. If we give the clients any other set of creds please let me know to update this.